docs/source/speechlm2/datasets.rst

```diff
@@ -0,0 +1,406 @@
+2. A user turn containing the audio and a text context (from the ``context`` tag)
+3. An assistant turn containing the transcription (from the cut's supervision text)
+
+If a ``context`` tag is provided in the configuration, it's added as a text turn before the audio.
+
+SALMDataset
+***********
+
+This dataset class is specialized for the SALM model, which focuses on understanding speech input and generating text output.
+
+.. code-block:: python
+
+    from nemo.collections.speechlm2.data import SALMDataset
+    
+    dataset = SALMDataset(
+        tokenizer=model.tokenizer,                   # Text tokenizer
+    )
+
+DataModule
+----------
+
+The DataModule class in the speechlm2 collection manages dataset loading, preparation, and batching for PyTorch Lightning training:
+
+.. code-block:: python
+
+    from nemo.collections.speechlm2.data import DataModule
+    
+    datamodule = DataModule(
+        cfg_data,                  # Configuration dictionary for data
+        tokenizer=model.tokenizer, # Text tokenizer
+        dataset=dataset            # Instance of DuplexS2SDataset or SALMDataset
+    )
+
+The DataModule takes care of:
+1. Setting up proper data parallel ranks for dataloaders
+2. Instantiating the dataloaders with configuration from YAML
+3. Managing multiple datasets for validation/testing
+
+Bucketing for Efficient Training
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The DataModule supports bucketing for more efficient training. Bucketing groups samples of similar lengths together, which reduces padding and improves training efficiency. The key bucketing parameters are:
+
+1. **batch_duration**: Target cumulative duration (in seconds) of samples in a batch
+2. **bucket_duration_bins**: List of duration thresholds for bucketing
+3. **use_bucketing**: Flag to enable/disable bucketing
+4. **num_buckets**: Number of buckets to create
+5. **bucket_buffer_size**: Number of samples to load into memory for bucket assignment
+
+Example bucketing configuration:
+
+.. code-block:: yaml
+
+    train_ds:
+      # ... other settings
+      batch_duration: 100  # Target 100 seconds per batch
+      bucket_duration_bins: [8.94766, 10.1551, 11.64118, 19.30376, 42.85]  # Duration thresholds
+      use_bucketing: true  # Enable bucketing
+      num_buckets: 5  # Create 5 buckets
+      bucket_buffer_size: 5000  # Buffer size for bucket assignment
+
+When bucketing is enabled:
+
+1. Samples are grouped into buckets based on their duration
+2. Each batch contains samples from the same bucket
+3. The actual batch size can vary to maintain a consistent total duration
+4. The target batch_duration ensures efficient GPU memory usage
+
+Bucketing helps to:
+- Reduce padding and increase effective batch size
+- Improve training efficiency and convergence
+- Manage memory usage with variable-length inputs
+
+Data Configuration
+-----------------
+
+A typical data configuration in YAML includes:
```

Comment:
Isn't this specifc to S2S (and not SALM)?The flow may be a bit confusing. Also, S2S discussion includes an example manifest but SALM discussion does not. 

Does the following structure make sense? (i) add an example of manifest for SALM (or add a link to a relevant reference if this is already discussed elsewhere), (ii) add an example of yaml for s2s in the s2s section, (iii) Under "preparing dataset" mention that this is just for S2S and either add or link to the corresponding discussion for SALM.

Reply:
I'll remove this section in datasets.rst as I just notuiced it's duplicated with configs.rst